### PR TITLE
GH Issues: replace markdown-based Bug Report form w/ YAML-based form

### LINKED ISSUE #44: Tooling: Improve Grafana bug reports using Issue Forms beta
Improve Grafana bug reports using Issue Forms beta:
https://github.blog/changelog/2021-06-23-issues-forms-beta-for-public-repositories/

### PR DESCRIPTION
**Pull request by zuchka**
closes https://github.com/grafana/grafana-community-support-squad/issues/44

As discussed [on Slack here](https://raintank-corp.slack.com/archives/C03E2AQKU2W/p1681368733266319), the community squad is ready to test a new Bug Report form in production. We hope that this will reduce the need to collect more information from users.

[you can test it here](https://github.com/grafana/github-actions-testrepo/issues/new?assignees=&labels=&template=0-bug-report.yaml&title=Product-Area-Here%3A+short+description+of+bug+here)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        - I use Grafana Cloud
+        - Docker
+        - Kubernetes
+        - A package manager (APT, YUM, BREW, etc.)
+        - A downloaded binary
+        - Other
+        - I don't know
+    validations:
+      required: false
+  - type: dropdown
```
Comment (sakjur) [inline .github/ISSUE_TEMPLATE/0-bug-report.yaml:206]: Without versions, this information is only semi-useful. Can we make the OS field a free text form?

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    id: user-browser
+    attributes:
+      label: User's Browser?
+      description: Is the bug occuring in Grafana's user-interface? If so, what browsers are you seeing the problem on? You may choose more than one.
+      multiple: true
+      options:
+        - Firefox
+        - Chrome
+        - Safari
+        - Microsoft Edge
```
Comment (sakjur) [inline .github/ISSUE_TEMPLATE/0-bug-report.yaml:238]: Nit: No other browsers have been stylized with the corporate name, if we want to be consistent about this, we should probably also expand Google Chrome and Mozilla Firefox.

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+        - OpenSUSE
+        - Rocky
+        - Linux Mint
+        - Arch Linux
+        - Raspberry Pi OS
+        - Other Linux distro
+        - Other (non-Linux)
+        - I don't know
+  - type: dropdown
+    id: user-browser
```
Comment (sakjur) [inline .github/ISSUE_TEMPLATE/0-bug-report.yaml:229]: Same here as for the OS, the version matters a lot, I’d prefer a free text form to make sure we can get that.

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+        - Clickhouse
+        - Cloudwatch
+        - Datadog
+        - Dynatrace
+        - Elasticsearch
+        - Gitlab
+        - Google Cloud Monitoring
+        - Graphite
+        - Honeycomb
+        - InfluxDB
```
Comment (sakjur) [inline .github/ISSUE_TEMPLATE/0-bug-report.yaml:278]: Missing the image renderer here.

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+    id: gf-version
+    attributes:
+      label: |
+        
+        # 3 / 3: What Grafana version are you using?
+        
+      description: |
+        #
+        - You can find your Grafana version:
+          - inside the UI [here](https://share.cleanshot.com/wVTvrKhy), [here](https://share.cleanshot.com/5qg3RKF1), or [here](https://share.cleanshot.com/988MQZz1), depending on your platform and access level.
```
Comment (sakjur) [inline .github/ISSUE_TEMPLATE/0-bug-report.yaml:93]: I’d suggest placing all of this in the docs instead and having a single link to “how to figure out your version”. Linking to screenshots makes for poor accessibility and confusing interface.

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+          - using the Grafana HTTP API's `health` endpoint [like this](https://share.cleanshot.com/H7gcj2Lj): `curl your-grafana-url/api/health` 
+          - or running the `grafana-server -v` command from Grafana's root directory [like this](https://share.cleanshot.com/LHXfmtZx): `./bin/grafana-server -v`
+      multiple: false
+      options:
+        - I don't know my version
+        - Whatever Grafana Cloud is running
+        - Building from `main`
+        - 9.5.0
```
Comment (sakjur) [inline .github/ISSUE_TEMPLATE/0-bug-report.yaml:101]: This should probably be generated from our releases page or a free text form?

**Review by sakjur — COMMENTED** [review]
I think there are some minor problems that together amounts to a reduction in precision that might make the information gathering loop for issues longer.

I’d suggest loosening up the dropdowns to freetext fields before merging and see if that has a negative impact on the quality of the input compared to the maintenance burden and imprecision of dropdowns.

**Comment by sakjur** [comment]
Updated milestone to 10.0.x since 9.5.x would imply a backport which is unnecessary for this PR 🙂

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml (hunk 6 above)
Comment (natellium) [reply .github/ISSUE_TEMPLATE/0-bug-report.yaml:101]: I think we can start with a free text form. Looks like a pain to maintain right now :)

**Review by natellium — APPROVED** [review]
Agree with that may be easier to start with some free text forms instead and see what people fill in. But overall very cool!

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml (hunk 1 above)
Comment (zuchka) [reply .github/ISSUE_TEMPLATE/0-bug-report.yaml:206]: love it. absolutely. will move towards required freetext fields

File: .github/ISSUE_TEMPLATE/0-bug-report.yaml (hunk 6 above)
Comment (zuchka) [reply .github/ISSUE_TEMPLATE/0-bug-report.yaml:101]: agree. system right now is manual and brittle. free-text probably best place to start

**Comment by zuchka** [comment]
thanks @natellium and @sakjur for the great feedback. Migrated to open text fields. Much improved; less hard-coded stuff now, easier to maintain/read, more flexible. ❤️

**Pull request merged**